common/src/main/java/com/wynntils/features/user/QuestInfoOverlayFeature.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.mojang.blaze3d.platform.Window;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.overlays.Overlay;
+import com.wynntils.core.features.overlays.OverlayPosition;
+import com.wynntils.core.features.overlays.annotations.OverlayInfo;
+import com.wynntils.core.features.overlays.sizes.GuiScaledOverlaySize;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.mc.render.HorizontalAlignment;
+import com.wynntils.mc.render.TextRenderTask;
+import com.wynntils.mc.render.VerticalAlignment;
+import com.wynntils.utils.objects.CommonColors;
+import com.wynntils.wc.utils.scoreboard.quests.QuestInfo;
+import com.wynntils.wc.utils.scoreboard.quests.QuestManager;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QuestInfoOverlayFeature extends UserFeature {
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay QuestInfoOverlay = new QuestInfoOverlay();
+
+    public static class QuestInfoOverlay extends Overlay {
+        public QuestInfoOverlay() {
+            super(
+                    new OverlayPosition(
+                            5,
+                            -5,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Right,
+                            OverlayPosition.AnchorSection.TopRight),
+                    new GuiScaledOverlaySize(300, 50),
+                    HorizontalAlignment.Left,
+                    VerticalAlignment.Middle);
+        }
+
+        @Override
+        public void render(PoseStack poseStack, float partialTicks, Window window) {
+            QuestInfo currentQuest = QuestManager.getCurrentQuest();
+
+            if (currentQuest == null) {
+                return;
+            }
+
+            List<TextRenderTask> toRender = new ArrayList<>();
+
+            toRender.add(TextRenderTask.getWithHorizontalAlignment(
+                    "Tracked Quest Info:", this.getWidth(), CommonColors.GREEN, this.getRenderHorizontalAlignment()));
+            toRender.add(TextRenderTask.getWithHorizontalAlignment(
+                    currentQuest.quest(), this.getWidth(), CommonColors.ORANGE, this.getRenderHorizontalAlignment()));
+            toRender.add(TextRenderTask.getWithHorizontalAlignment(
+                    currentQuest.description(),
+                    this.getWidth(),
+                    CommonColors.WHITE,
+                    this.getRenderHorizontalAlignment()));
```

Comment:
Be careful with this pattern, this is going to generate a bunch of garbage and significantly reduce the time between each Young GC causing lag spikes. A good rule for rendering is to always try splitting logic from the actual rendering code,since whatever you place inside the render block will run each frame!

Reply:
I can extract toRender and update the text, so there are no new instances if it. I will do this later.

Reply:
Actually, I can't extract it as it can change. @HeyZeer0 do you have something in mind to optimize this as this is already pretty optimized imo?

Reply:
The problem with making them static is not the strings, but the overlay alignment changes.

Reply:
If the user changes configuration you mean?

Reply:
> I thought we said that we should have a way for features to get a callback when a config change is happening. That will probably need to be extended to cover overlays, too.

Yeah, but we don't have that right now